components/nodemanager-service/pgdb/node-managers-process-node.go

```diff
@@ -55,98 +48,28 @@ func (db *DB) ProcessIncomingNode(node *manager.NodeMetadata) error {
 	if timestamp.After(tenMinAgo) {
 		nodeState = manager.NodeState_RUNNING.String()
 	}
-	// TODO: give the target config for nodes table a default value in db
-	// so we don't have to do this.
-	tcByte, err := json.Marshal(nodes.TargetConfig{})
-	if err != nil {
-		return errors.Wrap(err, "ProcessIncomingNode unable to marshal target config")
-	}
-	lastContact := ptypes.TimestampString(node.LastContact)
-	// the incoming node may hit any of these cases:
-	// 1) it is already registered in our db with same uuid as report: update the node entry
-	// 2) it is already registered in our db with diff uuid, same source_id: update the node by source_id
-	// 3) it is not in our db, we must add it
-	// TODO (@vj): revisit this to address race condition issue; either decide on a code guard
-	// against that case, or put it all in sql
 
-	// here we address case #1
-	rows := int64(0)
-	if node.Uuid != "" {
-		updated, err := updateNode(node.Uuid, node, lastContact, db, nodeState)
-		if err != nil {
-			return errors.Wrapf(err, "ProcessIncomingNode error updating node %s", node.Uuid)
-		}
-		rows, err = updated.RowsAffected()
-		if err != nil {
-			return errors.Wrap(err, "ProcessIncomingNode unable to read rows affected")
-		}
-	}
-	if rows > 0 {
-		logrus.Debugf("updated node %s with last_contact %s", node.Uuid, node.LastContact)
-		return nil
-	}
-	if node.SourceId != "" {
-		// here we address case #2
-		err := upsertNodeBySourceID(node, lastContact, db, nodeState, tcByte)
-		if err != nil {
-			return errors.Wrapf(err, "ProcessIncomingNode unable to upsert node by source_id %s", node.SourceId)
-		}
-	} else {
-		// here we address case #3
-		err := addNodeToDB(node, lastContact, db, nodeState, tcByte)
-		if err != nil {
-			// TODO:
-			// if we get here and there is a duplicate node id error, we probably attempted to send a report in
-			// on a terminated node. we should check and log that msg if it is the case
-			return errors.Wrapf(err, "ProcessIncomingNode unable to add node to db %s", node.Name)
-		}
-	}
-	return nil
-}
+	lastContact := ptypes.TimestampString(node.LastContact)
 
-func updateNode(nodeUuid string, node *manager.NodeMetadata, lastContact string, db *DB, nodeState string) (sql.Result, error) {
-	updated, err := db.Exec(sqlUpdateNodeLastContact, nodeState, lastContact, node.JobUuid, node.Name, node.PlatformName, node.PlatformRelease, nodeUuid)
-	if err != nil {
-		return nil, errors.Wrapf(err, "updateNode unable to update node %s", node.Uuid)
+	if len(node.GetUuid()) == 0 {
+		node.Uuid = uuid.Must(uuid.NewV4()).String()
```

Comment:
We might want to return an error if we can not create the UUID instead of exiting the service.

Reply:
ahhh ya i forgot about that darn uuid must functionality, good catch - ya!